assets/js/modules/search-console/components/dashboard/DashboardWidgetPopularKeywordsTable.js

```diff
@@ -39,85 +35,74 @@ import Layout from '../../../../components/layout/layout';
 import {
 	isDataZeroSearchConsole,
 } from '../../util';
+import { STORE_NAME } from '../../datastore/constants';
+const { useSelect } = Data;
 
-class DashboardWidgetPopularKeywordsTable extends Component {
-	static renderLayout( component ) {
-		return (
-			<div className="
-				mdc-layout-grid__cell
-				mdc-layout-grid__cell--span-6-desktop
-				mdc-layout-grid__cell--span-4-tablet
-			">
-				<Layout
-					className="googlesitekit-popular-content"
-					footer
-					footerCtaLabel={ _x( 'Search Console', 'Service name', 'google-site-kit' ) }
-					footerCtaLink={
-						sprintf( 'https://search.google.com/search-console?resource_id=%s', getModulesData()[ 'search-console' ].settings.propertyID )
-					}
-					fill
-				>
-					{ component }
-				</Layout>
-			</div>
-		);
-	}
+const DashboardWidgetPopularKeywordsTable = ( props ) => {
+	const { data } = props;
+	const domain = useSelect( ( select ) => select( STORE_NAME ).getPropertyID() );
+	const baseServiceURL = useSelect( ( select ) => select( STORE_NAME ).getServiceURL( { path: '/performance/search-analytics' } ) );
+	const searchConsolePropertyMainURL = useSelect( ( select ) => select( STORE_NAME ).getServiceURL( { query: { resource_id: domain } } ) );
 
-	render() {
-		const { data } = this.props;
+	if ( ! data || ! data.length ) {
+		return null;
+	}
 
-		if ( ! data || ! data.length ) {
-			return null;
-		}
+	const headers = [
+		{
+			title: __( 'Top search queries for your site', 'google-site-kit' ),
+			tooltip: __( 'Most searched for keywords related to your content', 'google-site-kit' ),
+			primary: true,
+		},
+		{
+			title: __( 'Clicks', 'google-site-kit' ),
+			tooltip: __( 'Number of times users clicked on your content in search results', 'google-site-kit' ),
+		},
+		{
+			title: __( 'Impressions', 'google-site-kit' ),
+			tooltip: __( 'Counted each time your content appears in search results', 'google-site-kit' ),
+		},
+	];
+	const links = [];
 
-		const headers = [
-			{
-				title: __( 'Top search queries for your site', 'google-site-kit' ),
-				tooltip: __( 'Most searched for keywords related to your content', 'google-site-kit' ),
-				primary: true,
-			},
-			{
-				title: __( 'Clicks', 'google-site-kit' ),
-				tooltip: __( 'Number of times users clicked on your content in search results', 'google-site-kit' ),
-			},
-			{
-				title: __( 'Impressions', 'google-site-kit' ),
-				tooltip: __( 'Counted each time your content appears in search results', 'google-site-kit' ),
-			},
+	const dataMapped = data.map( ( row, i ) => {
+		const query = row.keys[ 0 ];
+		links[ i ] = addQueryArgs( baseServiceURL, { query, resource_id: domain, num_of_days: 28 } );
```

Comment:
`domain` is not part of `data` so it would make sense to put that `resource_id` query argument into the `baseServiceURL` above right away instead of handling it here for every link.

Reply:
Updated. I also moved `num_of_days` as well for the same reason.